kafka/producer_test.go

```diff
@@ -320,6 +374,37 @@ func TestProducerOAuthBearerConfig(t *testing.T) {
 	}
 
 	p.Close()
+	err = p.SetOAuthBearerToken(OAuthBearerToken{
```

Comment:
For a lot of the producer tests, the code looks like

```
func() {
 // 1. handling for non-nil, non-closed producer
 // 2. handling for closed producer
 // 3. handling for nil producer
}
```

in most of these cases, while 1 is too different from 2 and 3, 2 and 3 are mostly the same except the value of the producer. It would be better if they can be made into a separate function and called with different value of producer to cut down on code duplication.

Reply:
Same goes for the changes in TestProducerAPIs

Reply:
got it.